src/coreclr/jit/morph.cpp

```diff
@@ -10917,77 +10943,398 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
+                            }
+                        }
+                        else
+                        {
+                            assert(simdBaseType == TYP_DOUBLE);
+
+                            if (vecCon->gtSimdVal.u64[0] == 0x8000000000000000)
+                            {
+                                lhs = op2;
+                                op  = GT_NEG;
+                            }
+                        }
+                    }
+                }
+
+                if ((op == GT_NONE) && op2->IsVectorConst())
+                {
+                    GenTreeVecCon* vecCon = op2->AsVecCon();
+
+                    if (vecCon->IsBroadcast(retType, simdBaseType))
+                    {
+                        if (simdBaseType == TYP_FLOAT)
+                        {
+                            if (vecCon->gtSimdVal.u32[0] == 0x80000000)
+                            {
+                                lhs = op1;
+                                op  = GT_NEG;
+                            }
+                        }
+                        else
+                        {
+                            assert(simdBaseType == TYP_DOUBLE);
+
+                            if (vecCon->gtSimdVal.u64[0] == 0x8000000000000000)
+                            {
+                                lhs = op1;
+                                op  = GT_NEG;
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (op == GT_NONE)
             {
                 break;
             }
 
-            var_types    simdType        = node->TypeGet();
-            CorInfoType  simdBaseJitType = node->GetSimdBaseJitType();
-            unsigned int simdSize        = node->GetSimdSize();
+            // No need to filter side effects since we only have 1 operand
+            assert(rhs == nullptr);
 
-            GenTree* andnNode = gtNewSimdBinOpNode(GT_AND_NOT, simdType, lhs, rhs, simdBaseJitType, simdSize);
+            GenTree* resNode = gtNewSimdUnOpNode(op, retType, lhs, simdBaseJitType, simdSize);
 
             DEBUG_DESTROY_NODE(node);
-            INDEBUG(andnNode->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
+            INDEBUG(resNode->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
 
-            return andnNode;
+            return resNode;
         }
+#endif // TARGET_ARM64
+
         default:
         {
             break;
         }
     }
 
+    switch (oper)
+    {
+        case GT_ADD:
+        case GT_AND:
+        case GT_OR:
+        case GT_XOR:
+        {
+            GenTree* op1 = node->Op(1);
+            GenTree* op2 = node->Op(2);
+
+            if (op1->IsVectorConst())
+            {
+                // Commute constants to the right.
+                std::swap(op1, op2);
+
+                node->Op(1) = op1;
+                node->Op(2) = op2;
+            }
+
+            // TODO-CQ: Mirror the optimizations done by fgOptimizeCommutativeArithmetic
```

Comment:
The main thing missing here is the changing of `(op (COMMA(... (op X C1))) C2)" to "(COMMA(... (op X C3)))` and `((x + icon1) + (y + icon2))" to ((x + y) + (icon1 + icon2))` for integers

There's also some minor optimizations for add/mul like `-a + b to b - a`, `a + -b to a - b`, and `(~x + 1) to -x`